### DISCUSSION TITLE
How to retrieve all possible values of an attribute in metadata

### DISCUSSION THREAD

**Discussion by OweysMomenzada** [qdrant/qdrant#2721] [Q&A] [answered]
Is there a way to retrieve all possible values of a metadata? Or is that not possible?
Lets say I have the following metadata structure for all entries inside Qdrant (with three example entries):

```
{
 'page_content': <some text>.
  'metadata': {
      "date-of-upload": "2023-09-24",
      "language": "english",
      "page": 15,
      "source": "english/sensors-21-01322-v2.pdf"
    }
}


{
 'page_content': <some text>.
  'metadata': {
      "date-of-upload": "2023-07-14",
      "language": "english",
      "page": 11,
      "source": "english/sensors-21-01322-v2.pdf"
    }
}


{
 'page_content': <some text>.
  'metadata': {
      "date-of-upload": "2023-01-24",
      "language": "english",
      "page": 20,
      "source": "english/earning_calls.pdf"
    }
}
```

Now I want to now if there is a possibility to query all unique values for source:
`>>> ["english/earning_calls.pdf", "english/sensors-21-01322-v2.pdf"]`

**Comment by timvisee** [comment]
No. Qdrant does not support such feature to get unique values.

You could [scroll](https://qdrant.tech/documentation/concepts/points/#scroll-points) over all points, extract the desired metadata, and only keep unique values on your client. This would be very inefficient though.

**Comment by djptek** [comment, marked as answer]
Hi @OweysMomenzada 

I'd just like to update you on the availability of the [facet api ](https://qdrant.tech/documentation/concepts/payload/#facet-counts) which might fulfil your requirement

Here is a brief example of how that works

```
PUT collections/test
{
  "vectors": {
    "size": 3,
    "distance": "Dot"
  }
}

PUT /collections/test/index
{
    "field_name": "name",
    "field_schema": "keyword"
}


PUT collections/test/points
{
  "points": [
    {
      "id": 1,
      "vector": [255,0,0],
      "payload": {
        "name": "RED"
      }
    },
    {
      "id": 2,
      "vector": [0,255,0],
      "payload": {
        "name": "GREEN"
      }
    },
    {
      "id": 3,
      "vector": [0,0,255],
      "payload": {
        "name": "BLUE"
      }
    }
  ]
}

POST /collections/test/facet
{
    "key": "name"
}

### response

{
  "result": {
    "hits": [
      {
        "value": "BLUE",
        "count": 1
      },
      {
        "value": "GREEN",
        "count": 1
      },
      {
        "value": "RED",
        "count": 1
      }
    ]
  },
  "status": "ok",
  "time": 0.000356093
}
```

**Comment by FilippTrigub** [comment]
+1

having the same issue right now

**Reply by timvisee** [reply]
Please consider using the Facet API introduced in Qdrant 1.12.

Ref: https://github.com/orgs/qdrant/discussions/2721#discussioncomment-11567994

Docs: https://qdrant.tech/documentation/concepts/payload/#facet-counts
